app/models/destination.rb

```diff
@@ -1,5 +1,9 @@
 # frozen_string_literal: true
 
 class Destination < ActiveRecord::Base
-  validates :url, presence: true
+  enum service_type: {growth_spaces: "growth_spaces", adobe_campaigns: "adobe_campaigns"}
+
+  validates :url, :service_type, presence: true
+  validates :route_id, presence: true, if: :growth_spaces?
```

Comment:
I should think the `presence` of a `route_id` is also unconditional. Is there some use case where it might not be available?

Reply:
@hlbraddock `route_id` is always used as a parameter when calling the Growth Spaces API, so I assume it's a mandatory parameter, that's why I added the validation.
Do you think I should remove it?

Reply:
When I say that `route_id` is 'unconditional', I mean to say that it is **always** required, not just for `growth_spaces`.